modules/xfem/src/userobjects/GeometricCut3DUserObject.C

```diff
@@ -86,26 +106,26 @@ XFEMGeometricCut3D::cutElementByGeometry(const Elem * elem,
 }
 
 bool
-XFEMGeometricCut3D::cutFragmentByGeometry(std::vector<std::vector<Point>> & /*frag_edges*/,
-                                          std::vector<CutEdge> & /*cut_edges*/,
-                                          Real /*time*/)
+GeometricCut3DUserObject::cutFragmentByGeometry(std::vector<std::vector<Point>> & /*frag_edges*/,
+                                                std::vector<CutEdge> & /*cut_edges*/,
+                                                Real /*time*/) const
 {
-  mooseError("invalid method for 3D mesh cutting");
+  mooseError("Invalid method: must use vector of element faces for 3D mesh cutting");
   return false;
 }
 
 bool
-XFEMGeometricCut3D::cutFragmentByGeometry(std::vector<std::vector<Point>> & /*frag_faces*/,
-                                          std::vector<CutFace> & /*cut_faces*/,
-                                          Real /*time*/)
+GeometricCut3DUserObject::cutFragmentByGeometry(std::vector<std::vector<Point>> & /*frag_faces*/,
+                                                std::vector<CutFace> & /*cut_faces*/,
+                                                Real /*time*/) const
 {
   // TODO: Need this for branching in 3D
   mooseError("cutFragmentByGeometry not yet implemented for 3D mesh cutting");
   return false;
 }
 
 bool
-XFEMGeometricCut3D::intersectWithEdge(const Point & p1, const Point & p2, Point & pint)
+GeometricCut3DUserObject::intersectWithEdge(const Point & p1, const Point & p2, Point & pint) const
 {
   bool has_intersection = false;
   double plane_point[3] = {_center(0), _center(1), _center(2)};
```

Comment:
@dschwen, @bwspenc, and I spoke about this. The method interfaces with a system with various geometric operations taken into the XFEM module. We deemed changing this system to take moose types beyond the scope of the PR, but it is something that needs to be addressed in a future code cleanup PR.